content/troubleshooting/docs/reverse-proxy-sparkui.md

```diff
@@ -0,0 +1,290 @@
+driver_pod_name=$(kubectl get po -n emr | grep $job_id-driver | awk '{print $1}')
+# create a k8s service
+kubectl expose po -n emr \
+--port=4040 \
+--target-port 4040 \
+--name=$app_name-ui-svc \
+$driver_pod_name
+```
+The SparkUI service looks like this:
+```bash
+kubectl get svc -n emr
+
+NAME                  TYPE      CLUSTER-IP  EXTERNAL-IP   PORT(S)  AGE
+job-run-api-ui-svc ClusterIP 10.100.233.186  <none>      4040/TCP   9s
+```
+
+3.Finally, access the SparkUI in this format:
+```
+http://<YOUR_INGRESS_ADDRESS>/sparkui/<app_name>
+```
+Admin can get the ingress address by the CLI:
+```bash
+kubectl get ingress
+```
+## Launch EMR on EKS jobs by Spark Submit:
+
+1.Create an EMR on EKS pod with a service account that has the IRSA associated
+```bash
+kubectl run -it emrekspod \
+--image=public.ecr.aws/emr-on-eks/spark/emr-7.1.0:latest \
+--overrides='{ "spec": {"serviceAccount": "emr-containers-sa-spark"}}' \
+--command -n spark-operator /bin/bash
+```
+2.After login into the "emrekspod" pod, submit the job:
+```bash
+export app_name=sparksubmittest
+
+spark-submit \
+--master k8s://$KUBERNETES_SERVICE_HOST:443 \
+--deploy-mode cluster \
+--name $app_name \
+--class org.apache.spark.examples.SparkPi \
+--conf spark.ui.proxyBase=/sparkui/$app_name \
+--conf spark.ui.proxyRedirectUri="/" \
+--conf spark.kubernetes.container.image=public.ecr.aws/emr-on-eks/spark/emr-7.1.0:latest \
+--conf spark.kubernetes.authenticate.driver.serviceAccountName=emr-containers-sa-spark \
+--conf spark.kubernetes.namespace=spark-operator \
+local:///usr/lib/spark/examples/jars/spark-examples.jar 100000
+```
+3.As soon as the driver pod is running, create a kubernetes service for Spark UI and ensure its name has the suffix of **ui-svc**:
+```bash
+export app_name=sparksubmittest
+# get the running driver pod name
+driver_pod_name=$(kubectl get po -n spark-operator | grep -E 'sparksubmittest-.*-driver' | grep 'Running' | awk '{print $1}')
+
+# OPTIONAL - remove existing service if needed
+kubectl delete svc $app_name-ui-svc -n spark-operator
+
+# create k8s service
+kubectl expose po -n spark-operator \
+--port=4040 \
+--target-port 4040 \
+--name=$app_name-ui-svc \
+$driver_pod_name
+```
+
+## Appendix
+
+### deployment.yaml
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: spark-ui-reverse-proxy
+  labels:
+    app: spark-ui-reverse-proxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: spark-ui-reverse-proxy
+  template:
+    metadata:
+      labels:
+        app: spark-ui-reverse-proxy
+    spec:
+      containers:
+      - name: spark-ui-reverse-proxy
+        image: ghcr.io/datapunchorg/spark-ui-reverse-proxy:main-1652762636
+        imagePullPolicy: IfNotPresent
+        command:
+          - '/usr/bin/spark-ui-reverse-proxy'
+        args:
+          # EMR on EKS's namespace
+          - -namespace=spark-operator
```

Comment:
Alternatively, you can
```
env:
- name: NAMESPACE
  valueFrom:
      fieldRef:
          fieldPath: spec.metadata.namespace
args: ["$(NAMESPACE)"]
```

Reply:
We can't use this spec.metadata.namespace attribute, because the reverse proxy namespace is different from the listener namespace (EMR virtual cluster's namespace)